### PR TITLE
Updating interp codes for station points.

### PR DESCRIPTION
**Pull request by EricEngle-NOAA**
Updating all (non-spectral) interpolation codes to fully support interpolation from a supported source grid to stations points.

As of this commit, all existing tests pass and testing of interpolation to stations from grib2io are successful.

### CODE REVIEW

**Comment by edwardhartnett** [comment]
Do we have any tests for this code?

**Comment by EricEngle-NOAA** [comment]
I do not have tests for this at this time.  I *think* I should be able to add an interp to stations test inside tests/interp_mod_grib2.F90

**Comment by AlexanderRichert-NOAA** [comment]
Hi @EricEngle-NOAA I just added tests for station points which are now in develop. Can you merge those and see how the tests go? With maybe the exception of neighbor-budget interpolation, I think the stations points functionality is generally working (though that's based on limited testing).

**Comment by EricEngle-NOAA** [comment]
The latest couple of commits have fixed the failing/crashing station points tests.

**Comment by AlexanderRichert-NOAA** [comment]
That's great-- While we're at it, do you think it would be possible to enable the tests for the neighbor-budget scheme that are currently commented out?

**Comment by EricEngle-NOAA** [comment]
Yes.  I will work on that tomorrow.

**Comment by EricEngle-NOAA** [comment]
The neighbor-budget station tests have been implemented and are working.

File: src/neighbor_budget_interp_mod.F90 [hunk 1]
```diff
@@ -388,13 +398,16 @@ SUBROUTINE interpolate_neighbor_budget_vector(IPOPT,grid_in,grid_out, &
     ! - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
     !  COMPUTE NUMBER OF OUTPUT POINTS AND THEIR LATITUDES AND LONGITUDES.
     IRET=0
-    IF(.not. to_station_points) THEN
-       CALL GDSWZD(grid_out, 0,MO,FILL,XPTS,YPTS, &
-            RLON,RLAT,NO,CROT,SROT)
+    if(to_station_points) then
+       CALL GDSWZD(grid_out, 0,MO,FILL,XPTS,YPTS,RLON,RLAT,NO,CROT,SROT)
```
Comment (AlexanderRichert-NOAA) [inline src/neighbor_budget_interp_mod.F90]: Does this line need to be in both parts of the if/else block or can it be pulled out in front?

File: src/neighbor_interp_mod.F90 [hunk 2]
```diff
@@ -93,6 +93,7 @@ module neighbor_interp_mod
   !> - 3    unrecognized output grid
   !>
   !> @author Mark Iredell @date 96-04-10  
+  !> @author Eric Enlgle @date 23-05-04
```
Comment (AlexanderRichert-NOAA) [inline src/neighbor_interp_mod.F90]: name typo

**Comment by AlexanderRichert-NOAA** [comment]
This looks great, thanks @EricEngle-NOAA. If you can check on and resolve my last two comments then I'm happy to get this merged.

File: src/neighbor_budget_interp_mod.F90 (hunk 1 above)
Comment (EricEngle-NOAA) [reply src/neighbor_budget_interp_mod.F90]: Yes.  That can be pulled of the if block.  I will fix.

File: src/neighbor_interp_mod.F90 (hunk 2 above)
Comment (EricEngle-NOAA) [reply src/neighbor_interp_mod.F90]: Will fix.

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Pull request merged**